improver_tests/cube_combiner/test_MaxInTimeWindow.py

```diff
@@ -0,0 +1,184 @@
+
+def setup_realization_cubes() -> CubeList:
+    """Set up a period diagnostic cubes."""
+    realizations = [0, 1, 2, 3]
+    data = np.full((len(realizations), 2, 2), 0.1, dtype=np.float32)
+    times = [datetime(2017, 11, 10, hour) for hour in [3, 6, 9, 12]]
+    cubes = CubeList()
+    for increment, time in enumerate(times):
+        cubes.append(
+            set_up_variable_cube(
+                data + increment * 0.1,
+                name="lwe_thickness_of_precipitation_amount",
+                realizations=realizations,
+                spatial_grid="equalarea",
+                time=time,
+                time_bounds=(time - timedelta(hours=3), time),
+                frt=datetime(2017, 11, 9, 0),
+            )
+        )
+    cube = cubes.merge_cube()
+    return CubeList(cube.slices_over("time"))
+
+
+def setup_probability_cubes() -> CubeList:
+    """Set up a period diagnostic cubes."""
+    thresholds = [0, 1, 2, 3]
+    data = np.full((len(thresholds), 2, 2), 0.1, dtype=np.float32)
+    times = [datetime(2017, 11, 10, hour) for hour in [3, 6, 9, 12]]
+    cubes = CubeList()
+    for increment, time in enumerate(times):
+        cubes.append(
+            set_up_probability_cube(
+                data + increment * 0.1,
+                variable_name="lwe_thickness_of_precipitation_amount",
+                thresholds=thresholds,
+                spatial_grid="equalarea",
+                time=time,
+                time_bounds=(time - timedelta(hours=3), time),
+                frt=datetime(2017, 11, 9, 0),
+            )
+        )
+    cube = cubes.merge_cube()
+    return CubeList(cube.slices_over("time"))
+
+
+@pytest.mark.parametrize(
+    "forecast_cubes", [(setup_realization_cubes), (setup_probability_cubes)],
+)
+def test_basic(forecast_cubes):
+    """Test for max in a time window."""
+    expected = np.full((4, 2, 2), 0.4)
+    result = MaxInTimeWindow()(forecast_cubes())
+    intervals = (
+        "of lwe_thickness_of_precipitation_amount over 3.0 hours within time window"
+    )
+    cell_methods = (
+        CellMethod("sum", coords=["time"], intervals=intervals),
+        CellMethod("maximum", coords=["time"]),
+    )
+    if result.coords("lwe_thickness_of_precipitation_amount"):
+        assert (
+            result.name()
+            == "probability_of_lwe_thickness_of_precipitation_amount_above_threshold"
+        )
+    else:
+        assert result.name() == "lwe_thickness_of_precipitation_amount"
+    assert result.cell_methods == cell_methods
+    np.testing.assert_allclose(result.data, expected)
+
+
+def test_existing_cell_methods():
+    """Test the handling of existing cell methods."""
+    cubes = setup_realization_cubes()
+    intervals = "of lwe_thickness_of_precipitation_amount"
+    for cube in cubes:
+        cube.add_cell_method(CellMethod("sum", coords=["time"], intervals=intervals))
+        cube.add_cell_method(CellMethod("maximum", coords=["height"]))
+    result = MaxInTimeWindow()(cubes)
```

Comment:
Can you move your call to the plugin below the setting up of the expected values, which is what we tend to do everywhere else.

Reply:
I've adjusted this as suggested.